### ISSUE TITLE
homo nn: problem with sparse_categorical_crossentropy loss in v1.4.0

### ISSUE THREAD

**Issue by wang-boyu** [FederatedAI/FATE#1408]
**Describe the bug**
In v1.4.0 homo nn, when `softmax` activation and `sparse_categorical_crossentropy` loss are used, the performance of the model is extremely poor.

**To Reproduce**
1. Upload data

<details>
  <summary>Guest (click to expand)</summary>

```
{
    "file": "examples/data/breast_homo_guest.csv",
    "head": 1,
    "partition": 16,
    "work_mode": 1,
    "table_name": "homo_breast_guest",
    "namespace": "homo_breast_guest"
}
```
</details>

<details>
  <summary>Host (click to expand)</summary>

```
{
    "file": "examples/data/breast_homo_host.csv",
    "head": 1,
    "partition": 16,
    "work_mode": 1,
    "table_name": "homo_breast_host",
    "namespace": "homo_breast_host"
}
```
</details>

2. Submit job

<details>
  <summary>dsl (click to expand)</summary>

```
{
  "components": {
    "dataio_0": {
      "module": "DataIO",
      "input": {
        "data": {
          "data": [
            "args.train_data"
          ]
        }
      },
      "output": {
        "data": [
          "train"
        ],
        "model": [
          "dataio"
        ]
      }
    },
    "homo_nn_0": {
      "module": "HomoNN",
      "input": {
        "data": {
          "train_data": [
            "dataio_0.train"
          ]
        }
      },
      "output": {
        "data": [
          "train"
        ],
        "model": [
          "homo_nn"
        ]
      }
    },
    "homo_nn_1": {
      "module": "HomoNN",
      "input": {
        "data": {
          "eval_data": [
            "dataio_0.train"
          ]
        },
        "model": [
          "homo_nn_0.homo_nn"
        ]
      },
      "output": {
        "data": [
          "train2"
        ],
        "model": [
          "homo_nn2"
        ]
      }
    },
    "evaluation_0": {
      "module": "Evaluation",
      "input": {
        "data": {
          "data": [
            "homo_nn_1.train2"
          ]
        }
      },
      "output": {
        "data": [
          "evaluate"
        ]
      }
    }
  }
}
```
</details>

<details>
  <summary>conf (click to expand)</summary>

```
{
  "initiator": {
    "role": "guest",
    "party_id": 10000
  },
  "job_parameters": {
    "work_mode": 1
  },
  "role": {
    "guest": [
      10000
    ],
    "host": [
      9999
    ],
    "arbiter": [
      10000
    ]
  },
  "role_parameters": {
    "guest": {
      "args": {
        "data": {
          "train_data": [
            {
              "name": "homo_breast_guest",
              "namespace": "homo_breast_guest"
            }
          ]
        }
      },
      "dataio_0": {
        "with_label": [
          true
        ],
        "label_name": [
          "y"
        ],
        "label_type": [
          "int"
        ],
        "output_format": [
          "dense"
        ]
      }
    },
    "host": {
      "args": {
        "data": {
          "train_data": [
            {
              "name": "homo_breast_host",
              "namespace": "homo_breast_host"
            }
          ]
        }
      },
      "dataio_0": {
        "with_label": [
          true
        ],
        "label_name": [
          "y"
        ],
        "label_type": [
          "int"
        ],
        "output_format": [
          "dense"
        ]
      }
    }
  },
  "algorithm_parameters": {
    "homo_nn_0": {
      "config_type": "nn",
      "nn_define": [
        {
          "layer": "Dense",
          "units": 6,
          "activation": "relu"
        },
        {
          "layer": "Dense",
          "units": 2,
          "activation": "softmax"
        }
      ],
      "batch_size": -1,
      "optimizer": {
        "optimizer": "Adam",
        "learning_rate": 0.05
      },
      "early_stop": {
        "early_stop": "diff",
        "eps": 1e-4
      },
      "loss": "sparse_categorical_crossentropy",
      "metrics": [
        "accuracy"
      ],
      "max_iter": 5
    },
    "evaluation_0": {
      "eval_type": "binary"
    }
  }
}
```
</details>

3. Check data output and model performance

![softmax_data_output_v14](https://user-images.githubusercontent.com/10785873/82991928-0111b200-a031-11ea-92c2-5d94d7268e38.PNG)

![softmax_eval_v14](https://user-images.githubusercontent.com/10785873/82991996-17b80900-a031-11ea-9c91-600f7c847ef6.PNG)

**Expected behavior**
Performance should be consistent across different versions of FATE when the same algorithm are used on the same data.

**Screenshots**
As attached.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - Browser: Chrome

**Smartphone (please complete the following information):**
 - N.A.

**Additional context**
- In `v1.3.0`, with the same data, dsl, and conf, the results are like this:

![softmax_data_output_v13](https://user-images.githubusercontent.com/10785873/82992059-31595080-a031-11ea-9af6-068e9fc6e561.PNG)

![softmax_eval_v13](https://user-images.githubusercontent.com/10785873/82992074-361e0480-a031-11ea-8e87-5d4078bb6684.PNG)

**Comment by weiwee** [comment]
Thanks for the detailed report!  
We fix this issue with pr #1419 problem which will be released alone with next mirror version.

**Comment by wang-boyu** [comment]
Thanks! Looking forward to your next release.
